### PR TITLE
u3d/available: allow to match using regular expression

### PR DESCRIPTION
**Pull request by lacostej**
requires #215 to be merged. I will rebase when ready.

### CODE REVIEW

File: lib/u3d_core/core_ext/hash.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+## --- END LICENSE BLOCK ---
+module CoreExtensions
+  module Extractable
+    refine Hash do
+      def extract(*keys)
+        select { |k, _v| keys.include?(k) }
```
Comment (niezbop) [inline lib/u3d_core/core_ext/hash.rb]:
```ruby
def extract(*args)
  self.keys & args
end
```
This may be cleaner?

Comment (lacostej) [reply lib/u3d_core/core_ext/hash.rb]: does that work?

Comment (niezbop) [reply lib/u3d_core/core_ext/hash.rb]: Oops, maybe not. This should though:
```ruby
def extract(*args)
  self.keys & args
end
```

Comment (lacostej) [reply lib/u3d_core/core_ext/hash.rb]: image

Comment (niezbop) [reply lib/u3d_core/core_ext/hash.rb]: Misunderstood the intent. Sorry

**Pull request merged**